File: frontend/src/app/classes/entities/Reservation.ts
import { ReservationId } from "../ids/ReservationId";
import { Account } from "./Account";

export class Reservation
{
    id : ReservationId
    endDate : Date
    account : Account
    cancelled : boolean

    constructor(id: ReservationId, endDate : Date, account : Account, cancelled : boolean)
    {
        this.id = id;
        this.endDate = endDate;
        this.account = account;
        this.cancelled = cancelled;
    }
}

File: backend/src/main/java/com/tencheeduard/hotelapp/entities/Reservation.java
package com.tencheeduard.hotelapp.entities;

import com.tencheeduard.hotelapp.embeddables.ReservationId;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Date;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter @Setter
public class Reservation {

    @EmbeddedId
    ReservationId id;

    @Temporal(TemporalType.DATE)
    Date endDate;

    @ManyToOne
    Account account;

    Boolean cancelled;
}
